src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC } from 'react'
+import styled from 'styled-components'
+
+import COLORS from '../shared/colors'
+
+type SizeTypes = 'small' | 'medium' | 'large' | 'xlarge'
+
+type AvatarProps = {
+  size: SizeTypes
+  src?: string
+  alt?: string
+  id: string
+  name: string
+}
+
+const getTextSize = ({ size }: AvatarProps): number => {
+  switch (size) {
+    case 'small':
+      return 16
+    case 'medium':
+      return 20
+    case 'large':
+      return 36
+    case 'xlarge':
+      return 72
+    default:
+      return 20
+  }
+}
+
+const getCircleSize = ({ size }: AvatarProps): number => {
+  switch (size) {
+    case 'small':
+      return 32
+    case 'medium':
+      return 40
+    case 'large':
+      return 72
+    case 'xlarge':
+      return 144
+    default:
+      return 40
+  }
+}
+
+const getColorFromId = ({ id }: AvatarProps): string => {
+  const idAsNumber = parseInt(id, 10)
+  const color = idAsNumber % 4
+  switch (color) {
+    case 0:
+      return `${COLORS.razzleDazzleRose}`
+    case 1:
+      return `${COLORS.cornflowerBlue}`
+    case 2:
+      return `${COLORS.wildStrawberry}`
+    case 3:
+      return `${COLORS.salmon}`
+    default:
+      return `${COLORS.razzleDazzleRose}`
+  }
+}
+
+const Circle = styled.div<AvatarProps>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  overflow: hidden;
+
+  width: ${getCircleSize}px;
+  height: ${getCircleSize}px;
+  border-radius: 50%;
+  background-color: ${getColorFromId};
+`
+const Image = styled.img`
+  width: 100%;
+  height: 100%;
+`
+
+const Text = styled.p`
+  margin: 0;
+  font-size: ${getTextSize}px;
+  font-weight: 500;
+  font-family: Roboto, sans-serif;
+  color: ${COLORS.white};
+`
+
+const Avatar: FC<AvatarProps> = props => {
+  const { src, name, ...rest } = props
+  const initials = name
+    .split(' ')
+    .map(n => n[0])
+    .join('')
+
+  return (
+    <Circle {...props}>
```

Comment:
I don't think we should prop spread. Its better to define all props so you know exactly what props are handled. I think `className` prop should be added to `Circle` only and not on `Text` or a styled component will give both the same style

Reply:
Yea I partly agree, I mean, props is pretty good in this scenario since we can reuse the types in our styled components. Think of it, it's pretty neat.